### DISCUSSION TITLE
UniFi Wireguard

### DISCUSSION THREAD

**Discussion by pax0707** [netalertx/NetAlertX#575] [Q&A]
Both arp-scan (after adding VLANs) and Unifi Importer detect all my main, IOT, and Guest hosts fine but not those connecting to wireguard VPN server running on my UDM.
It looks like the discovery of VPN devices is not supported but wanted to check in on that to make sure.

**Comment by jokob-sk** [comment]
Hi @pax0707 ,

This is the crowd-sourced documentation on configuring subnets: https://github.com/jokob-sk/Pi.Alert/blob/main/docs/SUBNETS.md Have you checked that page? 

I'm not a wireguard expert so I asked ChatGPT:

> Arpscan (Address Resolution Protocol scan) is a tool used to discover hosts on a local network by sending ARP requests and analyzing the responses. However, WireGuard operates at a different layer of the OSI model compared to ARP. WireGuard operates at the network layer (Layer 3) for creating secure point-to-point connections over public networks, while ARP operates at the link layer (Layer 2) for mapping IP addresses to MAC addresses within a local network.
> 
> Since WireGuard does not rely on ARP for address resolution within its encrypted tunnels, using arpscan directly on a WireGuard interface wouldn't provide meaningful results. 

> Since WireGuard tunnels traffic over existing networks (such as the Internet or local networks), the devices within the WireGuard network are typically not directly visible or accessible from the outside. Each device on the WireGuard network is only aware of its direct peers, and there isn't a broadcast mechanism like ARP that would allow for easy discovery of all devices on the network.

Not sure if this is useful.

**Reply by pax0707** [reply]
I have checked Pi.Alert's docs and didn't help.

If AI is right, nothing to be done there.

Cheers.
